### PR TITLE
feat(optimizer)!: Alias expanded USING STRUCT fields

### PR DESCRIPTION
**Pull request by VaggelisD**
Fixes https://github.com/TobikoData/sqlmesh/issues/3465

### CODE REVIEW

File: sqlglot/optimizer/qualify_columns.py [hunk 1]
```diff
@@ -220,11 +220,18 @@ def _update_source_columns(source_name: str) -> None:
...
-                replacement = exp.func("coalesce", *coalesce_args)
+                replacement: exp.Expression = exp.func("coalesce", *coalesce_args)
 
-                # Ensure selects keep their output name
                 if isinstance(column.parent, exp.Select):
+                    # Ensure the USING column keeps it's output name if it's projected
                     replacement = alias(replacement, alias=column.name, copy=False)
+                elif not isinstance(column.parent, exp.PropertyEQ) and column.find_ancestor(
+                    exp.Struct
+                ):
```
Comment (georgesittas) [inline sqlglot/optimizer/qualify_columns.py:230]: Can't we just do `if isinstance(column.parent, exp.Struct)`?

Comment (VaggelisD) [reply sqlglot/optimizer/qualify_columns.py:230]: Yeah I think that simplifies the condition, also doesn't make a difference for nested structs

**Review by georgesittas — APPROVED** [review]
This needs to be marked as breaking. Is it universally correct to inject a `PropertyEQ` in that `Struct`?

**Comment by VaggelisD** [comment]
>  Is it universally correct to inject a PropertyEQ in that Struct?

Looks like it, tested a bunch of dialects and they all have `PropertyEq`. Also, by searching `exp.PropertyEq` I see that it comes up on STRUCT contexts around the main dialects.

Should we make it BQ specific for the time being?

**Pull request merged**